### DISCUSSION TITLE
Leverage Ansible default providers

### DISCUSSION THREAD

**Discussion by ryanmerolle** [PaloAltoNetworks/pan-os-ansible#207] [Ideas]
### Is your feature request related to a problem?

Feature Request

### Describe the solution you'd like

It would be great for these modules to leverage the applicable network_cli and/or httpapi ansible_connection mechanism.

### Describe alternatives you've considered

Leverage the current method of mapping `ansible_username` to the `provider.username` and `ansible_password` to the `provider.password` each time I call a panos task.

### Additional context

Not only would this further confirm with other network operating system ansible collections, you would alo be able to list this collection on https://docs.ansible.com/ansible/2.10/network/user_guide/platform_index.html

**Comment by welcome-to-palo-alto-networks** [comment]
:tada: Thanks for opening your first issue here! Welcome to the community!
